component/acme-dns.libsonnet

```
@@ -73,19 +41,69 @@ local scriptRole = kube.Role('acme-dns-secret-editor') {
   ],
 };
 
-local scriptRoleBinding = kube.RoleBinding('acme-dns-secret-editor') {
-  subjects_: [ scriptServiceAccount ],
-  roleRef_: scriptRole,
+local roleBinding = kube.RoleBinding('acme-dns-secret-editor') {
+  metadata: {
+    labels: {
+      name: 'acme-dns-secret-editor',
+    },
+    name: 'acme-dns-secret-editor',
+    namespace: params.namespace,
+  },
+  subjects_: [ serviceAccount ],
+  roleRef_: role,
 };
 
-local scriptPodSpec(name, script) = {
+local configMap = kube.ConfigMap('acme-dns-scripts') {
+  metadata: {
+    labels: {
+      name: 'acme-dns-scripts',
+    },
+    name: 'acme-dns-scripts',
+    namespace: params.namespace,
+  },
+  data: {
+    'register.sh': importstr 'scripts/acme-register.sh',
+    'check.sh': importstr 'scripts/acme-check.sh',
+  },
+};
+
+// Acme DNS Client
+
+local hasRegistrationSecret(name) = std.objectHas(acmeClients[name].api, 'username');
+local registrationSecret(name) = kube.Secret('acme-dns-%s-register' % name) {
+  metadata: {
+    labels: {
+      name: 'acme-dns-%s-register' % name,
+    },
+    name: 'acme-dns-%s-register' % name,
+    namespace: params.namespace,
+  },
+  stringData: {
+    REG_USERNAME: acmeClients[name].api.username,
+    REG_PASSWORD: acmeClients[name].api.password,
+  },
```

Since we've had issues when leaving the `data` field visible in other components (not sure why this hasn't been a problem here so far), I'd explicitly hide it here since we're already touching the code:

```suggestion
  },
  data:: {},
```